src/Yaapii.Atoms/Map/Map.Envelope.cs

```diff
@@ -52,7 +52,21 @@ public Envelope(Func<IDictionary<string, string>> origin, bool live)
                 this.fixedOrigin = new Sticky<IDictionary<string, string>>(origin);
             }
 
-            public string this[string key] { get => Val()[key]; set => throw this.rejectWriteExc; }
+            public string this[string key]
+            {
+                get
+                {
+                    try
+                    {
+                        return Val()[key];
+                    }
+                    catch (KeyNotFoundException)
+                    {
+                        throw new ArgumentException($"The key '{key}' is not present in the map.");
```

Comment:
It would be helpful if we had the possible keys in the exception message. Of course this makes sense only for string keys (so the first two ofmour envelopes).

Reply:
Tests are using `int` keys and the exception message is showing the key in all three cases (`Object.ToString()`). Because of that I also added the possible ToString-Keys for the non-string-key envelope. Worst case: showing generic 'some object' ToString implementation.